envbuilder.go

```diff
@@ -604,9 +604,14 @@ func Run(ctx context.Context, options Options) error {
 
 	// Remove the Docker config secret file!
 	if options.DockerConfigBase64 != "" {
-		err = os.Remove(filepath.Join(MagicDir, "config.json"))
-		if err != nil && !os.IsNotExist(err) {
-			return fmt.Errorf("remove docker config: %w", err)
+		c := filepath.Join(MagicDir, "config.json")
+		err = os.Remove(c)
+		if err != nil {
+			if !os.IsNotExist(err) {
+				return fmt.Errorf("remove docker config: %w", err)
+			} else {
+				fmt.Fprintln(os.Stderr, "failed to remove the Docker config secret file: %w", c)
```

Comment:
Why not use `options.Logger` here and warn?

Reply:
I saw both options being used in the code so I didn't know what to use, to be honest.

Reply:
That's a fair point. I'll raise a separate PR to standardize this.